### ISSUE TITLE
consumer: lots empty messages in kafka live data

### ISSUE THREAD

**Issue by Lingling7** [spicavigo/kafka-rust#57]
I deployed a consumer as the example, I came across lots empty msgs, but in our production kafka live data there are no empty messages. Is there anyone could help me figure this out? Many thanks!

**Comment by Lingling7** [comment]
I checked online to see if there are some explanation, I found a possible answer:” Note also that we ask for a fetchSize of 100000 bytes. If the Kafka producers are writing large batches, this might not be enough, and might return an empty message set. In this case, the fetchSize should be increased until a non-empty set is returned.” [link] (https://cwiki.apache.org/confluence/display/KAFKA/0.8.0+SimpleConsumer+Example)

I checked the producer, there are no empty data, so I think I should increase the fetch size based on the size of producer write batch. How can I change the size through the kafka-rust code?

**Comment by xitep** [comment]
hello, i encountered this problem, but i'm still know sure about the right solution. in the meantime you could try `branch-0.2.x` which allows you to [set the max fetch size](https://github.com/spicavigo/kafka-rust/blob/branch-0.2.x/src/client.rs#L427) to overcome the problem. You can use this in your `Cargo.toml`:

```
[dependencies.kafka]
git = "https://github.com/spicavigo/kafka-rust.git"
branch = "branch-0.2.x"
```

instead of

```
[dependencies]
kafka = "*"
```

**Comment by Lingling7** [comment]
@xitep thank you so much

**Comment by xitep** [comment]
@Lingling7 I have tried to address the issue you reported as part of the new `Consumer` (see `branch-0.2.x`). Now, the consumer's `.poll()` method will return an error by default when it discovers the unlucky situation.

I believe the right solution is by specifying a reasonable `fetch_max_bytes_per_partition` value on the `KafkaClient` instance, since most of the time one's consumers should be properly set up correspondingly to the producers in one's landscape. However, I can imagine a situation where one just might exceptionally want to retry with a bigger "fetch size" for the sake of one or two big messages in the topic. For exactly these situations, I have provided the `.with_retry_max_bytes_limit` setting on the consumer.

The change is prepared as part of #64 and is very likely to be merged tomorrow.

**Comment by xitep** [comment]
It seems like failing is not a good option when being subscribed to multiple partitions since a partition might not be able to write its data fast enough to the response, while the response itself might be satisfied with data from another (depending on `KafkaClient::fetch_min_bytes` setting).  I'll drop throwing the error in the consumer therefore.

**Comment by xitep** [comment]
I've finally found a reliable way to signal the error even while consuming multiple partitions. The fix - #68 -  will be part of the `Consumer` in the `0.2.0` release.

**Comment by xitep** [comment]
just pushed `kafka-rust:0.2.0` to crates.io with the mentioned fix. feel free to re-open if the problem occurs again.
